### PR TITLE
handle sync PUT errors

### LINKED ISSUE #149: Uncaught TypeError: Cannot read property 'length' of null
@bsclifton reports this error in 0.17.7 in Slack:
```
(BLESSED_EXTENSION context for cjnmeadmgmiihncdidmfiabhenbggfjm) chrome-extension://cjnmeadmgmiihncdidmfiabhenbggfjm/content/scripts/sync.js:4: Uncaught TypeError: Cannot read property 'length' of null{TypeError: Cannot read property 'length' of null
    at Object.utf8_length [as length] (chrome-extension://cjnmeadmgmiihncdidmfiabhenbggfjm/content/scripts/sync.js:4:9091)
    at Writer.write_string [as string] (chrome-extension://cjnmeadmgmiihncdidmfiabhenbggfjm/content/scripts/sync.js:49:5596)
    at Function.encode (chrome-extension://cjnmeadmgmiihncdidmfiabhenbggfjm/content/scripts/sync.js:2:30717)
    at Function.encode (chrome-extension://cjnmeadmgmiihncdidmfiabhenbggfjm/content/scripts/sync.js:3:5348)
    at Function.encode (chrome-extension://cjnmeadmgmiihncdidmfiabhenbggfjm/content/scripts/sync.js:2:23081)
    at Serializer.syncRecordToByteArray (chrome-extension://cjnmeadmgmiihncdidmfiabhenbggfjm/content/scripts/sync.js:4:3726)
    at Object.module.exports.encrypt (chrome-extension://cjnmeadmgmiihncdidmfiabhenbggfjm/content/scripts/sync.js:1:1780)
    at RequestUtil.syncRecord [as encrypt] (chrome-extension://cjnmeadmgmiihncdidmfiabhenbggfjm/content/scripts/sync.js:1:1621)
    at RequestUtil.put (chrome-extension://cjnmeadmgmiihncdidmfiabhenbggfjm/content/scripts/sync.js:1:15736)
    at chrome-extension://cjnmeadmgmiihncdidmfiabhenbggfjm/content/scripts/sync.js:3:29020}
```

### PR DESCRIPTION
**Pull request by diracdeltas**
fix #149

### CODE REVIEW

**Comment by bsclifton** [comment]
I'm new to this code, but I wonder if it would make sense to check here too?
https://github.com/brave/sync/blob/fix/149/client/requestUtil.js#L231
(ex: if record is falsey, noop)

**Comment by ayumi** [comment]
@bsclifton that sounds reasonable – maybe `console.log` a warning and return a Promise which resolves immediately

**Comment by diracdeltas** [comment]
@bsclifton i think if record is falsey it should throw an error because that shouldn't happen. in that case, the caller catches the error and it's a no-op anyway.

**Comment by diracdeltas** [comment]
also if record is falsey, i think https://github.com/brave/sync/pull/150/files#diff-6ce8437e6c27185736d22013ce441e15R135 will throw an error and bufferedPut won't be called in the first place

File: client/sync.js [hunk 1]
```diff
@@ -131,14 +131,20 @@ const startSync = (requester) => {
     logSync(`Sending ${records.length} records`)
     const categoryId = proto.categories[category]
     records.forEach((record) => {
+      if (!record) {
+        return
```
Comment (ayumi) [inline client/sync.js]: should we be logging a warning here? falsey record is not to be expected

**Comment by diracdeltas** [comment]
the issue @bsclifton experienced seems to be due to our version of protobufjs not accepting `null` as a valid value for a `string` proto field. dealing with that now.

**Comment by diracdeltas** [comment]
working on test brokenness

File: client/recordUtil.js [hunk 2]
```diff
@@ -301,13 +303,24 @@ module.exports.syncRecordAsJS = (record) => {
...
   object.objectData = type
-  const data = record[type].toObject({defaults: true, enums: Number, longs: Number})
+  const data = api.SyncRecord[syncTypes[type]].toObject(record[type], {
+    defaults: true,
+    enums: Number,
+    longs: Number
+  })
   if (data.fields && data.fields.length > 0) {
     object[type] = pickFields(data, data.fields)
+    // Empty the fields field for backwards compatibility with protobufjs 6.6
```
Comment (diracdeltas) [inline client/recordUtil.js]: @ayumi is this needed? the `fields` field seems to always be empty with protobufjs 6.6 but it is populated with the correct fields in 6.8. however, it is expected to be empty for the tests to pass.

Comment (ayumi) [inline client/recordUtil.js]: i don't remember the details. what i do remember is pbjs didn't distinguish between props with unset default value versus fields explicitly set with that default value

so I added fields to explicitly track which fields were set.

**Pull request merged**